.github/workflows/sanity-test.yaml

```diff
@@ -0,0 +1,244 @@
+ # Onedrive test
+
+      # AWS creds
+      - name: Configure AWS credentials from Test account
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE }}
+          role-session-name: integration-testing
+          aws-region: us-east-1
+
+      # run the tests
+      - name: Backup onedrive test
+        id: onedrive-test
+        run: |
+          set -euo pipefail
+          ./corso backup create onedrive --user "${CORSO_M365_TEST_USER_ID}" --hide-progress 2>&1 | tee $TEST_RESULT/backup_onedrive.txt 
+          valid_num=$(grep -i -e 'Completed (0 errors)' $TEST_RESULT/backup_onedrive.txt | wc -l)
+
+          if [ $valid_num -ne 1 ]
+          then
+              echo "backup was not successfull"
+              exit 1
+          fi
+
+          echo result=$(grep -i -e 'Completed (0 errors)' $TEST_RESULT/backup_onedrive.txt  | awk '{print $2}') >> $GITHUB_OUTPUT
+
+      # list the bakcup onedrive
+      - name: Backup onedrive list test
+        run: |
+          set -euo pipefail
+          touch $TEST_RESULT/backup_onedrive_list.txt
+          ./corso backup list onedrive --hide-progress 2>&1 | tee $TEST_RESULT/backup_onedrive_list.txt 
+          valid_num=$(grep -i -e ${{ steps.onedrive-test.outputs.result }} $TEST_RESULT/backup_onedrive_list.txt | wc -l)
+
+          if [ $valid_num -ne 1 ]
+          then
+              echo "listing of backup was not successfull"
+              exit 1
+          fi
+
+      # AWS creds
+      - name: Configure AWS credentials from Test account
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE }}
+          role-session-name: integration-testing
+          aws-region: us-east-1
+
+      # test onedrive incremental
+      - name: Backup onedrive test
+        id: onedrive-incremental-test
+# TODO: check incremental time was smaller
+        run: |
+          set -euo pipefail
+          ./corso backup create onedrive --user "${CORSO_M365_TEST_USER_ID}" --hide-progress 2>&1 | tee $TEST_RESULT/backup_onedrive_incremental.txt 
+          valid_num=$(grep -i -e 'Completed (0 errors)' $TEST_RESULT/backup_onedrive_incremental.txt | wc -l)
+
+          if [ $valid_num -ne 1 ]
+          then
+              echo "backup was not successfull"
+              exit 1
+          fi
+
+          echo result=$(grep -i -e 'Completed (0 errors)' $TEST_RESULT/backup_onedrive_incremental.txt  | awk '{print $2}') >> $GITHUB_OUTPUT
+
+      # test onedrive restore
+      - name: Backup onedrive restore
+        id: onedrive-restore-test
+        run: |
+          set -euo pipefail
+          ./corso restore onedrive --backup "${{ steps.onedrive-incremental-test.outputs.result }}"  --hide-progress 2>&1 | tee $TEST_RESULT/onedrive-restore-test.txt 
+          echo result=$(grep -i -e 'Restoring to::' $TEST_RESULT/onedrive-restore-test.txt | awk -F "::" '{print $2}' )  >> $GITHUB_OUTPUT
```

Comment:
This might not be a problem, but I think splitting by `::` will cause it to take an extra space into the second field.